crates/ide/src/ide/inlay_hints.rs

```diff
@@ -0,0 +1,93 @@
+use crate::{DefDatabase, FileId};
+use itertools::Itertools;
+use syntax::{SyntaxKind, SyntaxToken, TextRange};
+
+#[derive(Debug, Clone)]
+pub enum InlayHintKind {
+    AttrsetAttribute(String),
+}
+
+#[derive(Debug, Clone)]
+pub struct InlayHintResult {
+    pub range: TextRange,
+    pub kind: InlayHintKind,
+}
+
+pub(crate) fn inlay_hints(
+    db: &dyn DefDatabase,
+    file: FileId,
+    range: Option<TextRange>,
+) -> Vec<InlayHintResult> {
+    let root_node = db.parse(file).syntax_node();
+
+    let (first_tok, end_pos) = match range {
+        None => (root_node.first_token(), u32::MAX.into()),
+        Some(range) => (
+            root_node.token_at_offset(range.start()).right_biased(),
+            range.end(),
+        ),
+    };
+
+    let hint_kind = |tok: &SyntaxToken| -> Option<InlayHintKind> {
+        if tok.kind() == SyntaxKind::SEMICOLON {
+            let attribute_node = tok
+                .parent()?
+                .first_child_by_kind(&|u: SyntaxKind| u == SyntaxKind::ATTR_PATH)?
+                .children()
+                .filter(|node| !node.kind().is_trivia());
+
+            let attr_name = attribute_node.map(|node| node.to_string()).join(".");
```

Comment:
With [`itertools::Itertools`](https://docs.rs/itertools/0.14.0/itertools/trait.Itertools.html#method.join), we can
```suggestion
            let attr_name = attribute_node.join(".");
```

It's already a dependency.